src/formats.c

```diff
@@ -905,6 +907,32 @@ static char *fmt_self_test_body(struct fmt_main *format,
 		}
 	}
 
+	if (format->params.flags & FMT_BLOB) {
+		/*
+		 * BLOB formats can't use the default binary_hash_[0-6] functions.
+		 */
+		if (format->methods.binary_hash[0] &&
+		    format->methods.binary_hash[0] ==
+		    fmt_default_binary_hash_0)
```

Comment:
No need for the `format->methods.binary_hash[0] &&` pre-check if you then check for equality to a specific non-NULL address.

Reply:
> No need for the `format->methods.binary_hash[0] &&` pre-check

That was from a copy/paste, I'll fix the originals as well.